### PR TITLE
New input configuration options

### PR DESCRIPTION
**Pull request by cgutman**
### Description
This PR adds a few input configuration options based on user requests.
- Gamepad autoselection logic can now be configured to include/exclude touchpad and motion sensor support when deciding on a controller type
- High resolution scroll events can now be disabled for applications that don't handle them properly (#1873)
- Native pen/touch support can now be disabled for applications without proper pen/touch support

I've also moved a couple input-related options from the General tab to the Input tab where they belong.

### Screenshot
image

image

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [x] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: docs/source/about/advanced_usage.rst [hunk 1]
```diff
@@ -237,6 +273,40 @@ always_send_scancodes
 
       always_send_scancodes = enabled
 
+high_resolution_scrolling
```
Comment (ReenigneArcher) [inline docs/source/about/advanced_usage.rst:276]: I wonder if we should think about having app specific settings. This seems like one example where that could be beneficial.

File: src_assets/common/assets/web/config.html [hunk 2]
```diff
@@ -818,6 +875,7 @@ <h2 class="accordion-header">
     "amd_vbaq": "enabled",
     "capture": "",
     "controller": "enabled",
+    "high_resolution_scrolling": "enabled",
     "install_steam_audio_drivers": "enabled",
     "ds4_back_as_touchpad_click": "enabled",
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/config.html:880]: Since you are touching this, can you alphabetize these keys?

**Review by ReenigneArcher — COMMENTED** [review]
Can you review this real quick as well? https://discord.com/channels/804382334370578482/1166059339170197574/1166059339170197574

When I made the note, I thought the misc button setting was missing from the UI and docs.

But looking closer at https://github.com/LizardByte/Sunshine/pull/1707 I'm don't think any UI change is required. Maybe just a note in the docs about the behavior of the share button?

**Comment by cgutman** [comment]
> But looking closer at https://github.com/LizardByte/Sunshine/pull/1707 I'm don't think any UI change is required. Maybe just a note in the docs about the behavior of the share button?

Yeah, there's no UI option to configure that.

File: docs/source/about/advanced_usage.rst (hunk 1 above)
Comment (cgutman) [reply docs/source/about/advanced_usage.rst:276]: We definitely should

File: src_assets/common/assets/web/config.html (hunk 2 above)
Comment (cgutman) [reply src_assets/common/assets/web/config.html:881]: Yep

**Review by ReenigneArcher — APPROVED** [review]

**Pull request merged**